pennylane/workflow/execution.py

```diff
@@ -535,7 +535,7 @@ def cost_fn(params, x):
     if (
         device_vjp
         and isinstance(device, qml.Device)
-        and "lightning" not in getattr(device, "short_name", "")
+        and "lightning" not in getattr(device, "name", "").lower()

```

Comment:
```suggestion
        and "lightning" not in getattr(device, "short_name", "").lower()
```
This is now for qpu and kokkos which still implement the old device interface.

Reply:
Though... I'm confused because we should still support VJPs for them through the `LightningVJPs` class?

Reply:
I chose `name` because LGPU and LK have a `name` attribute along the lines of `"Lightning ... plugin for PennyLane"`, so it still works, and we won't have to change this again once they're ported to the new API.

Reply:
> Though... I'm confused because we should still support VJPs for them through the LightningVJPs class?

This condition is for raising an error if we're requesting `device_vjp=True` with an old API device, and we don't want to include lightning devices in that list.